typing/types.mli

```diff
@@ -127,16 +127,21 @@ type type_desc =
 
   | Tunivar of string option
   (** Occurrence of a type variable introduced by a
-      forall quantifier / [Tpoly]. *)
+      forall quantifier / [tpoly]. *)
 
-  | Tpoly of type_expr * type_expr list
-  (** [Tpoly (ty,tyl)] ==> ['a1... 'an. ty],
+  | Tpoly of tpoly
+   (** [Tpoly {poly_body : ty; poly_params : tyl}] ==> ['a1... 'an. ty],
       where 'a1 ... 'an are names given to types in tyl
       and occurrences of those types in ty. *)
```

Comment:
In the documentation, I think it is better to use exclusively the new names rather than `ty` and `tyl`.